src/olympia/files/models.py

```diff
@@ -70,7 +70,9 @@ def files_upload_to_callback(instance, filename):
     """
     # Start with the add-on slug, but make it go through django's slugify() to
     # drop unicode characters.
-    name = slugify(instance.addon.slug).replace('-', '_') or 'addon'
+    name = (
+        slugify(instance.addon.slug or instance.addon.pk).replace('-', '_') or 'addon'
+    )
```

Comment:
maybe we should always fall back to the addon id, rather than `addon`?
```suggestion
    name = (
         (instance.addon.slug and slugify(instance.addon.slug).replace('-', '_')) or str(instance.addon.pk)
    )
```
?

Reply:
Yeah let's do that. d8607db353764e70a7dd0070505443630388ecad